sycl/include/sycl/detail/vector_convert.hpp

```diff
@@ -586,11 +533,17 @@ NativeToT convertImpl(NativeFromT Value) {
   else if constexpr (is_float_to_uint<FromT, ToT>::value)
     return ConvertFToU<NativeFromT, NativeToT, VecSize, ElemTy, RoundingMode>(
         Value);
-  else if constexpr (is_sint_to_uint<FromT, ToT>::value)
-    return SatConvertSToU<NativeFromT, NativeToT, VecSize, ElemTy>(Value);
   else {
-    static_assert(is_uint_to_sint<FromT, ToT>::value);
-    return SatConvertUToS<NativeFromT, NativeToT, VecSize, ElemTy>(Value);
+    static_assert(is_sint_to_from_uint<FromT, ToT>::value);
+    static_assert(VecSize == 1, "Conversion between signed and unsigned data "
+                                "types is only available for scalars");
+    // vec::convert is underspecified and therefore it is not entirely clear
+    // what to do here. 'static_cast' implementation matches SYCL CTS and it
+    // matches our old implementation. Unfortunately, OpSetConvertUToS and
+    // OpSatConvertSToU behave differently and we can't use them here until the
+    // behavior of conversions is well-defined by the SYCL 2020 speficiation.
```

Comment:
Nit: 
```suggestion
    // behavior of conversions is well-defined by the SYCL 2020 specification.
```

Reply:
Fixed in 2ba98e5bf276996a3083278ea416e4f2c2a3f89c